### ISSUE TITLE
Remove GPG binary deps

### ISSUE THREAD

**Issue by sapk** [go-gitea/gitea#13367]
We shouldn't rely on gpg binary for gpg operations and use go libs for that.

Because we will need to support various versions of gpg binary and more action to do for users.

Example: https://github.com/go-gitea/gitea/blob/fcb535c5c3b6b782d9242028fed4cd8c027c4e41/modules/git/repo_gpg.go#L18

Related issue: https://github.com/go-gitea/gitea/issues/10682

**Comment by zeripath** [comment]
Git requires gpg binary in any case.

**Comment by sapk** [comment]
@zeripath we shouldn't use the git binary also 😄.

**Comment by CirnoT** [comment]
@sapk While an interesting idea, it's not really viable to write our own implementation of Git

**Comment by sapk** [comment]
For git binary, we already have started to remove some part that were using the git cli by go-git lib but it is a very long running task and we should not go the same way with gpg.

**Comment by CirnoT** [comment]
Like I said, not feasible unless `go-git` receives actual funding from some big name company, look at compatibility chart, it doesn't even allow simple merges: https://github.com/go-git/go-git/blob/master/COMPATIBILITY.md

**Comment by lunny** [comment]
At least, this part could be replaced by Golang library. Let's remove GPG case by case. @zeripath

**Comment by CirnoT** [comment]
Please don't, there is no reason to spread resources and use different things in different places. We're not getting rid of GPG binary either way so there is absolutely no reason to waste time on this.

**Comment by CirnoT** [comment]
Additionally using library here would most likely spell doom for any use-case where user would want to keep commit signing key in agent, and not in unencrypted form on disk for Gitea to access. I feel like this is such a huge issue that it should be mentioned in separate comment.

**Comment by zeripath** [comment]
I agree with @cirno here it seems inefficient use of resources - won't block a PR doing it but would prefer work elsewhere, like making diff more efficient. It's worth noting that go-git did not turn out to be an efficient change and may have damaged gitea's reputation. So any changes should be carefully checked.

**Comment by lunny** [comment]
If there is a PR for this, we need a performance test.
